Code/EnginePlugins/ProcGenPlugin/Tasks/Implementation/Utils.cpp

```diff
@@ -35,14 +35,32 @@ void ezProcGenExpressionFunctions::ApplyVolumes(ezExpression::Inputs inputs, ezE
 
   const ezSimdVec4f* pInitialValues = inputs[3].GetPtr();
 
+  const ezSimdVec4f* pImgMode = inputs[5].GetPtr();
```

Comment:
You can assume that imgMode and refcolor are constants and only read the .x() value, same as we do for the tagsetindex above. If you really want that the mode and refcolor can vary for each point you would need to iterate through the input array like we do for the position here.

Reply:
Ok, I wasn't sure whether the VM might execute multiple pieces of the graph in the same call and thus have data change in between (ie when it switches from one volume to another).